core/modules/config.go

```diff
@@ -233,3 +236,14 @@ func (cfg ModuleCodegenConfig) Clone() *ModuleCodegenConfig {
 	cfg.AutomaticGitignore = &clone
 	return &cfg
 }
+
+type ModuleConfigClient struct {
+	// The generator the client uses to be generated.
+	Generator string `json:"generator"`
+
+	// The directory the client is generated in.
+	Directory string `json:"directory"`
+
+	// Whether the client is generated using the local SDK.
+	LocalLibrary *bool `json:"localLibrary,omitempty"`
```

Comment:
Probably worth more comments on all these one *explaining* what they mean - since these turns up as user-facing docs.

Reply:
Also, are these final? These are incredibly fiddly to change later if we do backwards compat, etc. I'd like to avoid changing them too much, so let's check in with @shykes on the design here.

Reply:
We agreed with Solomon to rename it `dev`, so I updated the PR with that changes :)

Reply:
Actually, using `dag.Version`, the SDK module itself could be able to know if that's a built version or not, that way we can get rid of this dev param and it's up to the SDK to adapt his behaviour